### ISSUE TITLE
User Input 

### ISSUE THREAD

**Issue by Manghi** [conwayste/conwayste#11]
Short Term Goal: Keyboard and Mouse

I've already started improvements on this in the ggez04 branch, PR #6 .
This will be tied into how the Game Modes effort goes. 
For now, I think this ticket should track designing an extensible user input interface so that we have less work to worry about later.

**Comment by Manghi** [comment]
I was digging through the docs and I didn't find a way to do this yet in ggez 0.4. SDL2 supports a lot of this natively, but I the APIs would need to be pulled into the `ggez::mouse` context.

SDL2 supports twenty system cursors via [SDL_CreateSystemCursor()](https://wiki.libsdl.org/SDL_CreateSystemCursor).

Beyond that, SDL2 supports customization via two approaches:

1) A bitmap-defined custom cursor which can only be black and white. This is performed via `See SDL_CreateCursor()` and `SDL_FreeCursor()`.

2) A custom-colored cursor obviously cannot leverage the above methods. Instead we need to disable the native cursor via `SDL_ShowCursor()` and then draw our desired cursor at the coordinates of the mouse. 

In summary, I think the following would is needed:
```C
SDL_CreateColorCursor
SDL_CreateCursor
SDL_CreateSystemCursor
SDL_FreeCursor
SDL_GetCursor
SDL_GetDefaultCursor
SDL_SetCursor
SDL_ShowCursor
```

None of this should be difficult to implement. I think I'll get this merged as a PR for the next ggez release; the commitments for ggez 0.5 are already set this year and this isn't on the road map.

This shouldn't be all that difficult anyway, maybe done in a night or two. Fun exercise really.

**Comment by Manghi** [comment]
I have something that builds and runs, but it isn't working properly though no errors are reported. I have a feeling the interlink with SDL2 is missing somehow. Asking for help from the ggez crew....

**Comment by Manghi** [comment]
- Keyboard and mouse are working
- ggez switched from sdl2 to winit
- New keyboard/mouse improvements in ui_widgets branch currently under review

Closing as done/irrelevant.
